packages/worker/src/sdk/users/users.ts

```diff
@@ -164,7 +164,7 @@ const buildUser = async (
 const validateUniqueUser = async (email: string, tenantId: string) => {
   // check budibase users in other tenants
   if (env.MULTI_TENANCY) {
-    const tenantUser = await tenancy.getTenantUser(email)
+    const tenantUser = (await tenancy.getTenantUser(email)) as User | undefined
```

Comment:
I think this should be  `as PlatformUserByEmail | null`, which is a new document that was introduced as a part of the user fixes. Best thing would be to add that directly to the function signature of `getTenantUser` like:
```ts
export const getTenantUser = async (identifier: string): Promise<PlatformUserByEmail | null> => {
```

Couple of side points:

1. The method signature isn't actually validated inside this function due to the usage of `doWithDB` which isn't TS. So inside the db function we could still say `return "hello"` and it wouldn't complain. If we converted `backend-core/src/db/index.js` to typescript we'd get validation that the function actually returns what we say it does. Something to do in a future PR. 

2. I'm more a fan of returning `undefined` than returning `null` in most cases like you have done with this cast, so we could update the exception handling in `getTenantUser` to just `return` instead of `return null` and change the signature to use `undefined` instead. In either case we can update the if statement below to
```ts
if (tenantUser && ...)
```
rather than the explicit null check. 

3. it would be most accurate to introduce an additional type inside `types/src/documents/users.ts`:
```ts
/**
 * doc id is user id
 */
export interface PlatformUserById extends Document {
  tenantId: string
}
```
and have the `getTenantUser` function look like:
```ts
export const getTenantUser = async (identifier: string): Promise<PlatformUserByEmail | PlatformUserById | null> => {
```
I try to stay away from using multi return types as much as possible as it puts additional logic on the consumers to determine which return type they have received. Usually it's easier to favour dedicated signatures. e.g. 
```ts
const getTenantUserByEmail = (): Promise<PlatformUserByEmail | null>  => {}
const getTenantUserById = (): Promise<PlatformUserById | null>  => {}
```

Reply:
Yeah, we also need to just finish the conversion and we will get everything out of the box.

On actually viewing the interface for `PlatformUserByEmail`, tbh `PlatformUserByEmail` and `PlatformUserById` don't feel right as types - for me this is where it starts to get to a point where we are just creating types to satisfy typescript and people unfamiliar with the codebase have no idea what they are - it's not actually mapped to our domain. 

The entity is a `PlatformUser`, where the fields like `tenantId` are realistically optional - the only difference was how we fetched them (by email/tenantId) they return the same object. Obviously a matter of preference but IMO you want to use unions/multi types to put more on the consumer, and make them code more defensively for values that can actually be `null`/`undefined`, unless you account for it in the function itself and guarantee that it cannot return those values.

Reply:
> rather than the explicit null check

Since this is loose equality it will catch both, which is nice